src/exceptions.rs

```diff
@@ -1068,6 +1071,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(feature = "gil-refs")]
```

Comment:
Is there a reason not to convert this test to use bound? It looks like `into_ref()` is the only deprecated piece here.

Reply:
This explicitly tests the `std::error::Error` impl, which the `Bound` api can't implement anymore.

Reply:
Hmm, do we need to be providing an impl of `Error` for `Bound<'_, T>`?

I think I missed this because this PR doesn't (yet?) but that impl behind a cfg, unless I really missed it.